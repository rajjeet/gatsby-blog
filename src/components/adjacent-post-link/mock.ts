export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
