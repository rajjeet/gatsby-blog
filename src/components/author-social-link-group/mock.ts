import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

