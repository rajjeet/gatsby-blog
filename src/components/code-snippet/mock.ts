import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
