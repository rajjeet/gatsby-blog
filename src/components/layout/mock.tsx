import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

