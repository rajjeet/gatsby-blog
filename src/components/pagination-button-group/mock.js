export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
