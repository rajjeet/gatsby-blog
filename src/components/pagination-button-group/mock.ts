export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}) => ({
  className: {}, currentPage, numOfPages, paginationSlug: '',
});
