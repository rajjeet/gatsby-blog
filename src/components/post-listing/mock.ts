import { createMockPosts } from '../../utils/testing';

export const makeProps = ({
  className = {},
  posts = createMockPosts,
  heading = 'Post Listing',
  numOfPages = 5,
  currentPage = 1,
  paginationSlug = '/sample_slug/',
} = {}) => ({
  className,
  posts,
  heading,
  numOfPages,
  currentPage,
  paginationSlug,
});
