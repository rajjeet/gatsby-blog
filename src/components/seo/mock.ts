export const makeProps = () => ({
  postData: '',
  postImage: null,
  isBlogPost: 1,
});
