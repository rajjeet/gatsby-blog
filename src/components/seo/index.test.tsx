import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Seo from './index';
import { makeProps } from './mock';

afterEach(cleanup);

describe('<Seo />', () => {
  it('should render', () => {
    const { asFragment } = render(<Seo {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
