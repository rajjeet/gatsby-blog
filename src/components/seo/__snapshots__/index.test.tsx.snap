// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Seo /> should render 1`] = `<DocumentFragment />`;
